src/pages/settings/Dunnings/Dunnings.tsx

```diff
@@ -159,77 +159,85 @@ const Dunnings = () => {
-                  </Typography>
-                ) : (
-                  <InfiniteScroll
-                    onBottom={() => {
-                      const { currentPage, totalPages } = data.dunningCampaigns.metadata
-
-                      currentPage < totalPages &&
-                        !loading &&
-                        fetchMore({
-                          variables: {
-                            page: currentPage + 1,
-                          },
-                        })
-                    }}
-                  >
-                    <Table
-                      name="dunnings-settings-list"
-                      containerSize={{ default: 0 }}
-                      rowSize={72}
-                      isLoading={loading}
-                      data={sortedTable}
-                      columns={[
-                        {
-                          key: 'name',
-                          title: translate('text_626162c62f790600f850b76a'),
-                          maxSpace: true,
-                          content: ({ name, code }) => (
-                            <div className="flex flex-1 items-center gap-3" data-test={code}>
-                              <Avatar size="big" variant="connector">
-                                <Icon size="medium" name="coin-dollar" color="dark" />
-                              </Avatar>
-                              <div>
-                                <Typography color="textSecondary" variant="bodyHl" noWrap>
-                                  {name}
-                                </Typography>
-                                <Typography variant="caption" noWrap>
-                                  {code}
-                                </Typography>
-                              </div>
-                            </div>
-                          ),
-                        },
-                      ]}
-                      actionColumnTooltip={() => translate('text_17285747264959xu1spelnh9')}
-                      actionColumn={(campaign) => {
-                        return [
-                          {
-                            startIcon: 'pen',
-                            title: translate('text_17321873136602nzwuvcycbr'),
-                            onAction: () => {
-                              navigate(
-                                generatePath(UPDATE_DUNNING_ROUTE, {
-                                  campaignId: campaign?.id || '',
-                                }),
-                              )
-                            },
-                          },
-                          {
-                            startIcon: 'trash',
-                            title: translate('text_1732187313660we30lb9kg57'),
-                            onAction: () => {
-                              deleteCampaignDialogRef.current?.openDialog(campaign)
+                )}
+
+                {hasAccessToFeature && (
```

Comment:
I feel like I see this logic of "has access to feature > display infinite scroll" a few times in the app. 

Should we consider using a component to handle this? Especially since the logic is also complex in the scolling